### PR TITLE
openstack nova usages plugin

### PR DESCRIPTION
**Pull request by mkurek**
- created base openstack plugin
- added new openstack nova instances plugin
- openstack ceilometer mysql and mongo plugins rewrited to use new base openstack plugin

### CODE REVIEW

**Comment by mkurek** [comment]
TODO:
- [ ] add unit tests to mongo plugin
- [ ] add unit tests to base plugin (from old ceilometer plugin)

File: src/ralph_scrooge/plugins/collect/_openstack_base.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        )[0]
+
+    @memoize
+    def get_daily_tenant(self, tenant_id, date):
+        try:
+            tenant = TenantInfo.objects.get(tenant_id=tenant_id)
+            return tenant.daily_tenants.get(date=date)
+        except TenantInfo.DoesNotExist as e:
+            raise TenantNotFoundError(e)
+        except DailyTenantInfo.DoesNotExist as e:
```
Comment (kula1922) [inline src/ralph_scrooge/plugins/collect/_openstack_base.py]: 'e' never used

File: src/ralph_scrooge/plugins/collect/_openstack_base.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                logger.error('Tenant {} not found'.format(tenant_id))
+                continue
+            DailyUsage(
+                date=date,
+                service_environment=daily_tenant.service_environment,
+                daily_pricing_object=daily_tenant,
+                value=value,
+                type=usage_type,
+                warehouse=warehouse,
+            ).save()
```
Comment (kula1922) [inline src/ralph_scrooge/plugins/collect/_openstack_base.py]: DailyUsage.create(...)

**Comment by kula1922** [comment]
No docstrings

File: src/ralph_scrooge/plugins/collect/_openstack_base.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        except TenantInfo.DoesNotExist as e:
+            raise TenantNotFoundError(e)
+        except DailyTenantInfo.DoesNotExist:
+            logger.warning('DailyTenant {} for date {} not found'.format(
+                tenant_id,
+                date,
+            ))
+            return tenant.get_daily_pricing_object(date)
+
+    def get_usages(self, date, connection_string):
```
Comment (kula1922) [inline src/ralph_scrooge/plugins/collect/_openstack_base.py]: It is not used enywhere else. In my opition this method is only for mysql but in OpenStackBasePlugin method get_usages schould raise "NotImplementedError".

**Comment by mkurek** [comment]
After merging this PR following script should be run:
```
from ralph_scrooge.models import UsageType
for ut in UsageType.objects.filter(name__startswith='openstack.'):
    ut.name = ut.name.replace('openstack.', 'openstack.ceilometer.')
    ut.symbol = ut.symbol.replace('openstack.', 'openstack.ceilometer')
    ut.save()
```
This will change "old" ceilometer usages names ("openstack....") to new convention for ceilometer ("openstack.ceilometer...")

**Comment by ghost** [comment]
LGTM

**Comment by ar4s** [comment]
:ok:

**Pull request merged**